mh/pat.py

```diff
@@ -1039,6 +1046,7 @@ def recvReqFmpInfo(self, packet_id, data, seq):
             self.sendAnsFmpInfo(fmp_data, fields, seq)
         elif packet_id == PatID4.ReqFmpInfo2:
             self.sendAnsFmpInfo2(fmp_data, fields, seq)
+        self.session.request_reconnection = True
```

Comment:
Maybe add a comment to explain why this is needed (i.e. to preserve the session at the server selection screen).